src/Kirel.Repositories/KirelGenericEntityFrameworkRepository.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Linq.Expressions;
+using Kirel.Repositories.Interfaces;
+using Kirel.Repositories.Persistance;
+using Kirel.Repositories.Sorts;
+using Microsoft.EntityFrameworkCore;
+
+namespace Kirel.Repositories.Infrastructure.Generics;
+
+/// <summary>
+/// Generalized Repository class type 
+/// </summary>
+public class KirelGenericEntityFrameworkRepository<TKey, TEntity> : IGenericRepository<TKey, TEntity> 
+    where TEntity : class, ICreatedAtTrackedEntity, IKeyEntity<TKey> 
+    where TKey :  IComparable, IComparable<TKey>, IEquatable<TKey>
```

Comment:
Please make TDbContext as type parameter, and not forget about where TDbContext : DbContext

Reply:
Added